Fix getJoinTableName for sqlite with schema attribute

mairo744 opened the pull request:
Incorrect table name for join tables in sqlite db when it contains schema attribute.
`getJoinTableName` doesnt depending on the platform.
Table name was "schema.table" instead of "schema__table".
Inspired by similar function `getTableName`.

Ocramius commented:
Please do add tests verifying your fix :+1:

ajgarlag commented:
I was suffering this problem, and this PR solved my issues. Can I help to get this PR merged?

greg0ire commented on tests/Doctrine/Tests/ORM/Functional/Ticket/DDC7079Test.php:
```diff
@@ -0,0 +1,126 @@
+    public function testGetTableName ()
+    {
+        $table = [
+            'name'=>'cms_user',
```
This does not look like it should be ok with our coding standard… maybe it changed since February? Can you please rebase, and take that as an occasion to squash all your commits into one commit?

Making sure you follow [this paragraph of the contributing guide](https://www.doctrine-project.org/contribute/index.html#working-on-topic-branches) when you do.

greg0ire replied:
As predicted, the build fails because of cs issues: https://travis-ci.org/doctrine/orm/jobs/607884252?utm_medium=notification&utm_source=github_status

Running `vendor/bin/phpcbf` should fix them.

ajgarlag replied:
@mairo744 I've just created a PR to your branch that fixes cs issues

mairo744 replied:
Thank you @greg0ire and @ajgarlag for help :)

greg0ire replied:
@ajgarlag it's quite rare to see people use git in a decentralized way. I love it when people really start to collaborate on things like this. Great job moving this forward!

greg0ire commented:
@ajgarlag I just added a new comment, so let's wait for OP to respond, and if there is no activity, you may open a new PR with these changes + my comment taken into account.

mairo744 committed:
Fix creation of join table names with schemas in SQLite

Join table name doesnt depending on the platform.
Table name was "schema.table" instead of "schema__table".

(cherry picked from commit 4878cd3f4ef30ffc6047c18e0f7b16aafeabc3b4)

greg0ire commented:
> I was suffering this problem, and this PR solved my issues. Can I help to get this PR merged?

A diagnosis regarding the build would help. It doesn't look related with this PR, but will have to be fixed. Maybe other PRs have the same issue?

ajgarlag commented:
@greg0ire test already fixed in #7889

greg0ire commented:
Ok so a simple rebase should fix things, right?

ajgarlag commented:
I think this PR should be added to next milestone `2.6.5`.

lcobucci commented:
@mairo744 @ajgarlag :ship:
